components/config.tsx

```diff
@@ -0,0 +1,50 @@
+export const config = {
```

Comment:
This config does not actually load from process.env in time, and will always end up assuming the variables don't exist.

To check I created a `.env.local` and changed the mnemonic and it didn't change.
If you remove the `|| "sell antenna drama rule twenty cement mad deliver you push derive hybrid"` then it does load up, so i'm assuming its a race condition

Reply:
pretty sure you don't want a MNEMONIC in anything "public"... we've struggled with this before

Reply:
especially with server side rendering... cc @acarnagey @ipbyrne any thoughts on this point?

We might want to open an issue to track a better solution for this.

Reply:
yeah - this is actually intentionally not at the `NEXT_PUBLIC_` prefix - if you do that stuff gets up in the fronted... All of these vars should stay backend and are working at that side

Reply:
> pretty sure you don't want a MNEMONIC in anything "public"... we've struggled with this before

@brownoxford and I were just discussing earlier splitting things out so that you can have an example mnemonic - effectively what is in there now - so that folks can try things out, that is separate from an actual mnemonic that could be treated as and used for actual signing and verifying.

Reply:
I published mnemonics intentionally in the front end and backend so that the dids were always the same... but normally you would never do that...

the backend would pull it from secrets, and the front end to generate in client side when requested by the user agent.

Reply:
you might also have a reversing index that looked like:

did:example:123#key-52 => mnemonic_123 hdpath_52

Reply:
@mkhraisha you can use these up at the fronted, you just want to be careful and do something like this:
```
import { config } from "../components/config";

export async function getServerSideProps(context: any) {
  var props = {
    config.theme_config
  };
  // DEBUG: console.log(props);
  return {
    props, // will be passed to the page component as props
  };
}
```

Reply:
> pretty sure you don't want a MNEMONIC in anything "public"... we've struggled with this before

Yeah, my understanding was this is the mnemonic that's initially hardcoded to the page on load, which is visible to the end-user.


> @mkhraisha you can use these up at the frontend, you just want to be careful and do something like this:
> 
> ```
> import { config } from "../components/config";
> 
> export async function getServerSideProps(context: any) {
>   var props = {
>     config.theme_config
>   };
>   // DEBUG: console.log(props);
>   return {
>     props, // will be passed to the page component as props
>   };
> }
> ```

awesome!

One thing I'll note however is the PR as is, does not actually provide a mechanism to modify the .env.local file with .env variables that would end up modifying the frontend.

I haven't tested yet with the snippet above, but as of now, it defaults all `process.env` variables to undefined on load time, which lets them take the default values rather than any value set by the user. 

To Test:
create a `.env.local` file, add `DID_MNEMONIC='RANDOM STRING HERE'' and you'll see it is not reflected in the UI.